packages/core/src/external/carequality/ihe-gateway-v2/outbound/xcpd/create/iti55-envelope.ts

```diff
@@ -45,152 +47,301 @@ function createSoapBody({
   const patientAddress = bodyData.patientResource.address?.[0];
   const patientTelecom = bodyData.patientResource.telecom?.[0]?.value ?? undefined;
 
-  const soapBody = {
-    "soap:Body": {
-      "@_xmlns:urn": namespaces.hl7,
-      "urn:PRPA_IN201305UV02": {
-        "@_ITSVersion": "XML_1.0",
-        "urn:id": {
-          "@_extension": messageId,
-          "@_root": homeCommunityId,
-        },
-        "urn:creationTime": {
-          "@_value": timestampToSoapBody(createdTimestamp),
-        },
-        "urn:interactionId": {
-          "@_extension": "PRPA_IN201305UV02",
-          "@_root": "2.16.840.1.113883.1.6",
-        },
-        "urn:processingCode": {
-          "@_code": "P",
-        },
-        "urn:processingModeCode": {
-          "@_code": "T",
-        },
-        "urn:acceptAckCode": {
-          "@_code": "AL",
-        },
-        "urn:receiver": {
-          "@_typeCode": "RCV",
-          "urn:device": {
-            "@_classCode": "DEV",
-            "@_determinerCode": "INSTANCE",
-            "urn:id": {
-              "@_root": receiverDeviceId,
-            },
-            "urn:telecom": {
-              "@_value": toUrl,
+  if (gateway.url !== specialNamespaceRequiredUrl) {
+    const soapBody = {
```

Comment:
Can we make functions that just returns the large payloads, like we had for this one before, and have `createSoapBody()` w/ logic? This helps reading it, its hard to keep track of the logic and these massive payloads.

Reply:
done!